doc/manual/modular-configurations.md

```diff
@@ -0,0 +1,429 @@
+---
+slug: modular-configurations
+---
+
+# Modular configurations
+
+This section explains how to leverage the Nickel languages features, and in
+particular the [merge system](./merging.md), to write reusable and maintainable
+configuration modules.
+
+## Programmable configuration
+
+One of the starting points of Nickel has been the limits of static data formats
+such as JSON, TOML or YAML when tackling large configurations. In particular,
+those formats have no way of reusing values of your configuration. Have a single
+value used in several places (say, an IP)? Have this virtual machine description
+block used several times, almost identically but for a few parameters? In JSON,
+the best you can do is to copy and paste, praying for the different copies to not
+drift away with time.
+
+In Nickel, you can just use variables and functions:
+
+```nickel
+let master_ip = "192.168.1.23" in
+let machine = fun {ip, open_ports} => {
+  # machine definition depending on ip and open_ports
+} in
+
+{
+  ip = master_ip,
+  cmd = "ssh -p 8070 user@%{master_ip}",
+  client1 = machine {ip = "0.0.0.0", open_ports = [22]},
+  client2 = machine {ip = "1.1.1.1", open_ports = [80]},
+}
+```
+
+In the zoology of programming abstractions, functions are the simplest and most
+ubiquitous concept of all. For the machine example, a ([pure][pure-function])
+function is exactly what we look for: an expression with unknown parameters
+which can be selected at usage time. Virtually every programming language has
+functions in some form, even if they're not called that way (for example, in
```

Comment:
```suggestion
functions in some form, even if they're not called that (for example, in
```

Reply:
I don't have a very strong opinion, but this part is verbatim from the blog which went under the review of the editing team. Is it a stylistic issue, or a correctness? If the former, I think I'll stick to the published version. If the latter, let me know, because then I need to update the blog post as well.

Reply:
It feels to me like "called that way" is about how something gets called, e.g. by phone. But I'm not a native speaker, so take that with a grain of salt. I don't think anyone will care, though :)